src/app/shared/ag-grid-editors/rating-editor/rating-editor.component.ts

```diff
@@ -0,0 +1,61 @@
+import { AfterViewInit, Component, ViewChild, ViewContainerRef } from '@angular/core';
+import { ICellEditorAngularComp } from 'ag-grid-angular';
+import { ICellEditorParams } from 'ag-grid';
+import { RATING_STATES_LABEL } from '../../../../constants/app-constants';
+
+@Component({
+  selector: 'app-rating-editor',
+  templateUrl: './rating-editor.component.html',
+  styleUrls: ['./rating-editor.component.scss']
+})
+export class RatingEditorComponent implements ICellEditorAngularComp, AfterViewInit {
+    value: string;
+    values: [ string ];
+    ratingStatesLabel = RATING_STATES_LABEL;
```

Comment:
we should also provide the rating labels when using the component, so as to make it generic